.github/workflows/docker.yml

```diff
@@ -0,0 +1,29 @@
+on:
+  push:
+    branches:
+    - main
+
+name: Release Docker Image
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+
+      - name: Install dependencies
+        run: yarn install --immutable --immutable-cache
+
+      - name: Build Docker Image
+        run: yarn docker build backend -t ghcr.io/${{ github.actor }}/rafiki-backend:${{ github.sha }}
+
+      - name: Login GHCR
+        # You must create a personal access token on GitHub and add it to the 
+        # repo's actions secrets named by 'GHCR_PAT' in order to use this step.
+        run: echo ${{ secrets.GHCR_PAT }} | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+
+      - name: Push Docker Image to GHCR
+        run: docker push ghcr.io/${{ github.actor }}/rafiki-backend:${{ github.sha }}
```

Comment:
Tagging with the git sha is fine, but I'd like to also have `latest` and/or `main`.
If we do want to also build and push images for git tags like I suggested above, I'd like to tag the image with tag version as well.
I'm trying to see if the [docker/metadata-action](https://github.com/docker/metadata-action) could be used for tagging even if we aren't using docker/build-push-action

Reply:
I added a conditional step to the workflow for tagging the image by tag name if the workflow is triggered by a tag push event.